sos/report/plugins/openstack_designate.py

```diff
@@ -0,0 +1,91 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, RedHatPlugin, UbuntuPlugin
+
+
+class OpenStackDesignate(Plugin):
+
+    short_desc = 'Openstack Designate'
+
+    plugin_name = "openstack_designate"
+    profiles = ('openstack', 'openstack_controller')
+
+    var_puppet_gen = "/var/lib/config-data/puppet-generated/designate"
+
+    def setup(self):
+        # configs
+        self.add_copy_spec([
+            "/etc/designate/*",
+            self.var_puppet_gen + "/etc/designate/designate.conf",
+            self.var_puppet_gen + "/etc/designate/pools.yaml",
+        ])
+
+        # logs
+        if self.get_option("all_logs"):
+            self.add_copy_spec([
+                "/var/log/designate/*",
+                "/var/log/containers/designate/*",
+            ])
+        else:
+            self.add_copy_spec([
+                "/var/log/designate/*.log",
+                "/var/log/containers/designate/*.log"
+            ])
+
+        # commands
+        self.add_cmd_output('openstack dns service list --all-projects')
+        self.add_cmd_output('openstack dns quota list --all-projects')
+        self.add_cmd_output('openstack zone list --all-projects')
+        self.add_cmd_output('openstack ptr record list --all-projects')
+        self.add_cmd_output('openstack zone export list --all-projects')
+        self.add_cmd_output('openstack zone import list --all-projects')
+        self.add_cmd_output('openstack zone blacklist list --all-projects')
+        self.add_cmd_output('openstack zone transfer request list '
+                            '--all-projects')
+        self.add_cmd_output('openstack zone transfer accept list '
+                            '--all-projects')
+        self.add_cmd_output('openstack tld list --all-projects')
```

Comment:
`add_cmd_output()` also accepts a list, and the preferred style in sos is to use a list when there are no additional parameters being passed, so let's condense these into a single call using a list of these commands.

After that, I believe this looks good now.